linera-core/src/chain_worker/state.rs

```diff
@@ -1379,11 +1379,12 @@ where
                 .await?;
         }
         if let Some(next_block_height) = query.test_next_block_height {
+            // If not, send the same error as if a block with next_block_height was proposed.
             ensure!(
                 chain.tip_state.get().next_block_height == next_block_height,
                 WorkerError::UnexpectedBlockHeight {
```

Comment:
Oh, this 3rd scenario for the error was indeed confusing because the block is not sent yet (IIRC).

Reply:
Yes, but as long as we can't add new error variants I think it's better to treat the chain info request as "the block" rather than the actual chain state on the validator.